nncf/experimental/post_training/algorithms/fast_bias_correction/algorithm.py

```diff
@@ -0,0 +1,312 @@
+        biased_nodes = nncf_graph.get_nodes_by_metatypes(layers_with_bias_types)
+
+        for node in biased_nodes:
+            node_name = node.node_name
+
+            if not self._is_node_with_bias(node):
+                nncf_logger.debug('Skipping node {} because there is no bias'.format(node_name))
+                continue
+            if not self._is_quantized_weights(node, nncf_graph):
+                nncf_logger.debug('Skipping node {} because weights was not quantized'.format(node_name))
+                continue
+
+            input_fp, input_shape = self._get_fp_inputs(statistic_points, node_name)
+            output_fp = self._get_fp_outputs(statistic_points, node_name)
+
+            input_name = node.layer_attributes.input_tensor_names[0]
+            output_name = node.layer_attributes.output_tensor_names[0]
+
+            extracted_model = self._extract_submodel(model_transformer, [input_name], [output_name])
+
+            channel_axis = self._backend_entity.channel_axis_by_types[node.node_type]
+            input_blob = self._create_input_blob(input_shape,
+                                                 input_fp,
+                                                 input_name)
+            bias_shift = self._get_bias_shift(
+                engine=engine,
+                model=extracted_model,
+                input_blob=input_blob,
+                channel_axis=channel_axis,
+                output_fp=output_fp,
+                output_name=output_name)
+
+            target_point = self._backend_entity.target_point(TargetType.LAYER,
+                                                             node.node_name)
+            bias_correction_command = self._backend_entity.bias_correction_command(target_point,
+                                                                                   bias_shift,
+                                                                                   self.threshold)
+            transformation_layout.register(bias_correction_command)
+
+        quantized_model = model_transformer.transform(transformation_layout)
+        return quantized_model
+
+    def _get_fp_inputs(self, statistic_points: StatisticPointsContainer, node_name: str) -> Tuple[List, List]:
+        """
+        Makes out per-layer needed data from the floating-point collected statistics
+
+        :param statistic_points: filled StatisticPointsContainer
+        :param node_name: name of the current layer
+        :return: collected mean tensor data and shape for the further bias calculation
+        """
+        def input_filter_func(point):
+            return PostTrainingAlgorithms.FastBiasCorrection in point.algorithm_to_tensor_collectors and \
+                point.target_point.type == TargetType.PRE_LAYER_OPERATION
+
+        input_fp = []
+        input_shape = []
+        for tensor_collector in statistic_points.iter_through_algorithm_tensor_collectors_in_target_node(
```

Comment:
`iter_through_algorithm_tensor_collectors_in_target_node()` name is really craze. `get_algo_statistics_for_node()`

Reply:
BTW, it can just return a list

Reply:
cc @kshpv, to clarify why we can't return the list here.